src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,106 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75F;
+    private static final int SIZE_MULTIPLIER = 2;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkGrow();
+        int index = calculateIndex(key);
+        if (table[index] == null) {
+            table[index] = createNode(key, value);
+        } else {
+            putInBucket(index, key, value);
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> currentNode = table[calculateIndex(key)];
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.key, key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private Node<K, V> createNode(K key, V value) {
```

Comment:
this method don't say that you increase size too.
Rename it 
```suggestion
    private Node<K, V> increaseSizeAndReturnNewNode(K key, V value) {
```